src/home/positions-list/components/position-controls/index.tsx

```diff
@@ -171,6 +179,26 @@ const PositionControls = ({
       position.to.address,
       position.chainId
     );
+  const account = web3Service.getAccount();
+  const companionAddress =
+    COMPANION_ADDRESS[position.version][position.chainId] || COMPANION_ADDRESS[LATEST_VERSION][position.chainId];
+
+  const isToProtocolToken = position.to.address === PROTOCOL_TOKEN_ADDRESS;
+  const hasYield = !!position.to.underlyingTokens.length || !!position.from.underlyingTokens.length;
+
+  const companionPermissions = createPermissionsObject(
+    position?.permissions?.find((permission) => permission.operator === companionAddress.toLowerCase())?.permissions
+  );
+
+  const accountPermissions = createPermissionsObject(
+    position?.permissions?.find((permission) => permission.operator === account.toLowerCase())?.permissions,
+    account.toLowerCase() === position.user.toLowerCase()
+  ) as PositionPermissions;
+
+  const mergedPermissions =
+    hasYield || isToProtocolToken
+      ? mergeCompanionPermissions(accountPermissions, companionPermissions)
```

Comment:
I'm not sure why we need to merge companion permissions based on some parameters